atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/client/creation/ClusterFactory.java

```diff
@@ -128,4 +136,40 @@ private static Cluster.Builder withLoadBalancingPolicy(
         // of less effective caching, default to TokenAwarePolicy.ReplicaOrdering.RANDOM childPolicy
         return builder.withLoadBalancingPolicy(new TokenAwarePolicy(policy));
     }
+
+    @Value.Immutable
+    public interface CassandraClusterConfig {
```

Comment:
this config terrifies me

Don't get me wrong, the original was also scary, but this has 4 ints that we can easily end up getting wrong (see or ask about "Crossed Wires" and the fun we had when passing two ints the wrong way around).

Consider making this a staged builder.